docs/vm-installation.md

```diff
@@ -0,0 +1,153 @@
+# Antrea Agent installation on VM
+
+Antrea Agent can run on a Linux or Windows VM, and enforce Antrea NetworkPolicies on the VM. This document describes
+the steps needed to configure and run `antrea-agent` on VMs.
+
+## Prerequisites on Kubernetes cluster
+
+1. Enable `ExternalNode` feature on the `antrea-controller`.
+2. Create a Namespace for `antrea-agent`. This document will use `vm-ns` as an example NameSpace for illustration.
+
+```bash
+kubectl create ns vm-ns
+```
+
+3. Create a ServiceAccount, ClusterRole and ClusterRoleBinding for `antrea-agent` as shown below. If you use a different
+   Namespace other than `vm-ns`, you need to update the [VM RBAC manifest](../build/yamls/externalnode/vm-agent-rbac.yml)
+   and change `vm-ns` to the right Namespace.
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/antrea-io/antrea/feature/externalnode/build/yamls/externalnode/vm-agent-rbac.yml
+```
+
+4. Create `antrea-agent.kubeconfig` file for `antrea-agent` to access the K8S API server.
+
+```bash
+export CLUSTER_NAME="kubernetes"
+export SERVICE_ACCOUNT="vm-agent"
+APISERVER=$(kubectl config view -o jsonpath="{.clusters[?(@.name==\"$CLUSTER_NAME\")].cluster.server}")
+TOKEN=$(kubectl -n vm-ns get secrets -o jsonpath="{.items[?(@.metadata.annotations['kubernetes\.io/service-account\.name']=='$SERVICE_ACCOUNT')].data.token}"|base64 --decode)
+kubectl config --kubeconfig=antrea-agent.kubeconfig set-cluster kubernetes --server=$APISERVER --insecure-skip-tls-verify=true
+kubectl config --kubeconfig=antrea-agent.kubeconfig set-credentials antrea-agent --token=$TOKEN
+kubectl config --kubeconfig=antrea-agent.kubeconfig set-context antrea-agent@kubernetes --cluster=kubernetes --user=antrea-agent
+kubectl config --kubeconfig=antrea-agent.kubeconfig use-context antrea-agent@kubernetes
+# Copy antrea-agent.kubeconfig to VM
+```
+
+5. Create `antrea-agent.antrea.kubeconfig` file for antrea-agent to access the `antrea-controller` API server.
+
+```bash
+# The VM should have network connectivity to reach antrea-controller server.
```

Comment:
This need not be nodeport IP, if the service can be exposed via LoadBalancer, then it would be a public IP. 
So should we say "expose `antrea-controller` service publicly so that it can be accessed outside the cluster?

Reply:
How about "# antrea-controller API server needs to be exposed via public IP or Node IP"

Reply:
I meant better to explain  what is the "172.18.0.1" IP or what IP it should be replaced with.

How about: "Specify the antrea-controller API server endpoint. antrea-controller needs to be exposed via the Node IP or a public IP that is reachable from the VM".